internal/ingress/controller/nginx.go

```diff
@@ -754,23 +754,31 @@ func (n *NGINXController) setupSSLProxy() {
 
 // IsDynamicConfigurationEnough decides if the new configuration changes can be dynamically applied without reloading
 func (n *NGINXController) IsDynamicConfigurationEnough(pcfg *ingress.Configuration) bool {
-	var copyOfRunningConfig ingress.Configuration = *n.runningConfig
-	var copyOfPcfg ingress.Configuration = *pcfg
+	copyOfRunningConfig := &ingress.Configuration{}
+	copyOfPcfg := &ingress.Configuration{}
+
+	mergo.MergeWithOverwrite(copyOfRunningConfig, n.runningConfig)
+	mergo.MergeWithOverwrite(copyOfPcfg, pcfg)
```

Comment:
is this safer? what was the issue with existing implementation?

Reply:
>  what was the issue with existing implementation?

No issue. We already use mergo for this kind of tasks